crates/wasmtime/src/config.rs

```diff
@@ -1728,6 +1728,23 @@ impl Config {
         self
     }
 
+    pub(crate) fn conditionally_enable_defaults(&mut self) {
+        // Enable tail calls by default when cranelift is going to be used as the compilation
+        // strategy, and we're not targeting s390x, which currently lacks tail-call support.
```

Comment:
Mind also editing this to cover that this logic is only applicable if tail calls were not otherwise explicitly disabled or enabled? (aka explaining the `.is_none()`)